src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidAdbCommand.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.DotNet.XHarness.Android;
+using Microsoft.DotNet.XHarness.CLI.CommandArguments.Android;
+using Microsoft.DotNet.XHarness.Common;
+using Microsoft.DotNet.XHarness.Common.CLI;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.XHarness.CLI.Commands.Android;
+
+internal class AndroidAdbCommand : XHarnessCommand<AndroidAdbCommandArguments>
+{
+    private const string Description = "Invoke bundled adb with given arguments";
+    protected override string CommandUsage { get; } = "android adb [OPTIONS] -- [ADB ARGUMENTS]";
+    protected override string CommandDescription => Description;
+
+    protected override AndroidAdbCommandArguments Arguments { get; } = new();
+
+    public AndroidAdbCommand() : base(TargetPlatform.Android, "adb", false, new ServiceCollection(), Description)
+    {
+    }
+
+    protected override Task<ExitCode> InvokeInternal(ILogger logger)
+    {
+        if (!PassThroughArguments.Any())
+        {
+            logger.LogError("Please provide delimeter '--' followed by arguments for ADB:" + Environment.NewLine +
+                $"    {CommandUsage}" + Environment.NewLine +
+                $"Example:" + Environment.NewLine +
+                $"    android adb --timeout 00:01:30 -- devices -l");
+
+            return Task.FromResult(ExitCode.INVALID_ARGUMENTS);
+        }
+
+        var runner = new AdbRunner(logger);
+
+        try
+        {
+            var result = runner.RunAdbCommand(string.Join(" ", PassThroughArguments), Arguments.Timeout);
+
+            Console.Write(result.StandardOutput);
+            Console.Error.Write(result.StandardError);
```

Comment:
This will write the output in different order if stdout and stderr were originally interleaved but maybe that's not a problem in this case?

Reply:
Yeah, I was thinking of adding some switch that would use only one StringBuilder to merge the two in the process manager but I tried calling adb.exe manually couple of times and the output it gave seemed to look fine enough like this. I would keep it like this and improve if there is some case where this is a problem (the code change was quite considerable)